### ISSUE TITLE
`gci-gke-alpha-features` test suite failing to bring up cluster

### ISSUE THREAD

**Issue by saad-ali** [kubernetes/kubernetes#69478]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:

https://k8s-testgrid.appspot.com/google-gke#gci-gke-alpha-features test suite has been failing since Sept 27. Failing to bring cluster up: 

```
Test Failures
BeforeSuite 10m6s
go run hack/e2e.go -v --test --test_args='--ginkgo.focus=BeforeSuite$'
_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:137
Oct  5 16:35:23.903: Error waiting for all pods to be running and ready: 2 / 19 pods in namespace "kube-system" are NOT in RUNNING and READY state in 10m0s
POD                                    NODE                                           PHASE   GRACE CONDITIONS
kube-dns-7df7d6c969-7cr6s              gke-e2e-32483-7506d-default-pool-eb898fb6-dpdq Running       [{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [kubedns sidecar]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [kubedns sidecar]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason: Message:}]
metrics-server-v0.3.1-66499c78f6-xcqgr gke-e2e-32483-7506d-default-pool-eb898fb6-dpdq Running       [{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:32:37 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [metrics-server]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:32:37 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [metrics-server]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-10-05 16:22:23 +0000 UTC Reason: Message:}]

_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:190

```

Example:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke-alpha-features/32491

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by saad-ali** [comment]
Failure is due to `kube-dns` and `metrics-server` containers not starting.

`kube-dns` is failing due to panic on startup:

```
I1005 16:35:23.300] Oct  5 16:35:23.300: INFO: Logs of kube-system/kube-dns-7df7d6c969-7cr6s:sidecar on node gke-e2e-32483-7506d-default-pool-eb898fb6-dpdq
I1005 16:35:23.301] Oct  5 16:35:23.300: INFO:  : STARTLOG
I1005 16:35:23.301] I1005 16:32:52.405697       1 main.go:51] Version v1.14.13
I1005 16:35:23.301] I1005 16:32:52.405808       1 server.go:45] Starting server (options {DnsMasqPort:53 DnsMasqAddr:127.0.0.1 DnsMasqPollIntervalMs:5000 Probes:[{Label:kubedns Server:127.0.0.1:10053 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:33} {Label:dnsmasq Server:127.0.0.1:53 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:33}] PrometheusAddr:0.0.0.0 PrometheusPort:10054 PrometheusPath:/metrics PrometheusNamespace:kubedns})
I1005 16:35:23.301] I1005 16:32:52.405858       1 dnsprobe.go:75] Starting dnsProbe {Label:kubedns Server:127.0.0.1:10053 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:33}
I1005 16:35:23.301] I1005 16:32:52.406395       1 dnsprobe.go:75] Starting dnsProbe {Label:dnsmasq Server:127.0.0.1:53 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:33}
I1005 16:35:23.301] panic: counter cannot decrease in value
I1005 16:35:23.301] 
I1005 16:35:23.301] goroutine 1 [running]:
I1005 16:35:23.302] k8s.io/dns/vendor/github.com/prometheus/client_golang/prometheus.(*counter).Add(0xc42043e0c0, 0xc055400000000000)
I1005 16:35:23.302] 	/go/src/k8s.io/dns/vendor/github.com/prometheus/client_golang/prometheus/counter.go:71 +0xb2
I1005 16:35:23.302] k8s.io/dns/pkg/sidecar.exportMetrics(0xc42000c120)
I1005 16:35:23.302] 	/go/src/k8s.io/dns/pkg/sidecar/server.go:86 +0x1b0
I1005 16:35:23.302] k8s.io/dns/pkg/sidecar.(*server).runMetrics(0xc42020d380, 0xc420066850)
I1005 16:35:23.302] 	/go/src/k8s.io/dns/pkg/sidecar/server.go:68 +0x1b9
I1005 16:35:23.302] k8s.io/dns/pkg/sidecar.(*server).Run(0xc42020d380, 0xc420066850)
I1005 16:35:23.302] 	/go/src/k8s.io/dns/pkg/sidecar/server.go:53 +0x29b
I1005 16:35:23.302] main.main()
I1005 16:35:23.302] 	/go/src/k8s.io/dns/cmd/sidecar/main.go:56 +0x1f0
I1005 16:35:23.302] 
I1005 16:35:23.302] ENDLOG for container kube-system:kube-dns-7df7d6c969-7cr6s:sidecar

```

`metrics-server` is failing due to timeout fetching `configmaps/extension-apiserver-authentication`:
```
I1005 16:35:23.725] Oct  5 16:35:23.725: INFO: Logs of kube-system/metrics-server-v0.3.1-66499c78f6-xcqgr:metrics-server on node gke-e2e-32483-7506d-default-pool-eb898fb6-dpdq
I1005 16:35:23.726] Oct  5 16:35:23.725: INFO:  : STARTLOG
I1005 16:35:23.726] Flag --deprecated-kubelet-completely-insecure has been deprecated, This is rarely the right option, since it leaves kubelet communication completely insecure.  If you encounter auth errors, make sure you've enabled token webhook auth on the Kubelet, and if you're in a test cluster with self-signed Kubelet certificates, consider using kubelet-insecure-tls instead.
I1005 16:35:23.726] I1005 16:32:06.109795       1 serving.go:273] Generated self-signed cert (apiserver.local.config/certificates/apiserver.crt, apiserver.local.config/certificates/apiserver.key)
I1005 16:35:23.726] W1005 16:32:36.542773       1 authentication.go:245] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
I1005 16:35:23.727] Error: Get https://10.63.240.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.727] Usage:
I1005 16:35:23.727]    [flags]
I1005 16:35:23.727] 
I1005 16:35:23.727] Flags:
I1005 16:35:23.727]       --alsologtostderr                                         log to standard error as well as files
I1005 16:35:23.727]       --authentication-kubeconfig string                        kubeconfig file pointing at the 'core' kubernetes server with enough rights to create tokenaccessreviews.authentication.k8s.io.
I1005 16:35:23.727]       --authentication-skip-lookup                              If false, the authentication-kubeconfig will be used to lookup missing authentication configuration from the cluster.
I1005 16:35:23.727]       --authentication-token-webhook-cache-ttl duration         The duration to cache responses from the webhook token authenticator. (default 10s)
I1005 16:35:23.728]       --authorization-kubeconfig string                         kubeconfig file pointing at the 'core' kubernetes server with enough rights to create  subjectaccessreviews.authorization.k8s.io.
I1005 16:35:23.728]       --authorization-webhook-cache-authorized-ttl duration     The duration to cache 'authorized' responses from the webhook authorizer. (default 10s)
I1005 16:35:23.728]       --authorization-webhook-cache-unauthorized-ttl duration   The duration to cache 'unauthorized' responses from the webhook authorizer. (default 10s)
I1005 16:35:23.728]       --bind-address ip                                         The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. If blank, all interfaces will be used (0.0.0.0 for all IPv4 interfaces and :: for all IPv6 interfaces). (default 0.0.0.0)
I1005 16:35:23.728]       --cert-dir string                                         The directory where the TLS certs are located. If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored. (default "apiserver.local.config/certificates")
I1005 16:35:23.729]       --client-ca-file string                                   If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.
I1005 16:35:23.729]       --contention-profiling                                    Enable lock contention profiling, if profiling is enabled
I1005 16:35:23.729]       --enable-swagger-ui                                       Enables swagger ui on the apiserver at /swagger-ui
I1005 16:35:23.729]   -h, --help                                                    help for this command
I1005 16:35:23.729]       --http2-max-streams-per-connection int                    The limit that the server gives to clients for the maximum number of streams in an HTTP/2 connection. Zero means to use golang's default.
I1005 16:35:23.729]       --kubeconfig string                                       The path to the kubeconfig used to connect to the Kubernetes API server and the Kubelets (defaults to in-cluster config)
I1005 16:35:23.729]       --kubelet-insecure-tls                                    Do not verify CA of serving certificates presented by Kubelets.  For testing purposes only.
I1005 16:35:23.729]       --kubelet-port int                                        The port to use to connect to Kubelets (defaults to 10250) (default 10250)
I1005 16:35:23.730]       --kubelet-preferred-address-types strings                 The priority of node address types to use when determining which address to use to connect to a particular node (default [Hostname,InternalDNS,InternalIP,ExternalDNS,ExternalIP])
I1005 16:35:23.730]       --log-flush-frequency duration                            Maximum number of seconds between log flushes (default 5s)
I1005 16:35:23.730]       --log_backtrace_at traceLocation                          when logging hits line file:N, emit a stack trace (default :0)
I1005 16:35:23.730]       --log_dir string                                          If non-empty, write log files in this directory
I1005 16:35:23.730]       --logtostderr                                             log to standard error instead of files (default true)
I1005 16:35:23.730]       --metric-resolution duration                              The resolution at which metrics-server will retain metrics. (default 1m0s)
I1005 16:35:23.730]       --profiling                                               Enable profiling via web interface host:port/debug/pprof/ (default true)
I1005 16:35:23.731]       --requestheader-allowed-names strings                     List of client certificate common names to allow to provide usernames in headers specified by --requestheader-username-headers. If empty, any client certificate validated by the authorities in --requestheader-client-ca-file is allowed.
I1005 16:35:23.731]       --requestheader-client-ca-file string                     Root certificate bundle to use to verify client certificates on incoming requests before trusting usernames in headers specified by --requestheader-username-headers. WARNING: generally do not depend on authorization being already done for incoming requests.
I1005 16:35:23.731]       --requestheader-extra-headers-prefix strings              List of request header prefixes to inspect. X-Remote-Extra- is suggested. (default [x-remote-extra-])
I1005 16:35:23.731]       --requestheader-group-headers strings                     List of request headers to inspect for groups. X-Remote-Group is suggested. (default [x-remote-group])
I1005 16:35:23.731]       --requestheader-username-headers strings                  List of request headers to inspect for usernames. X-Remote-User is common. (default [x-remote-user])
I1005 16:35:23.731]       --secure-port int                                         The port on which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS at all. (default 443)
I1005 16:35:23.731]       --stderrthreshold severity                                logs at or above this threshold go to stderr (default 2)
I1005 16:35:23.732]       --tls-cert-file string                                    File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert). If HTTPS serving is enabled, and --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory specified by --cert-dir.
I1005 16:35:23.732]       --tls-cipher-suites strings                               Comma-separated list of cipher suites for the server. If omitted, the default Go cipher suites will be use.  Possible values: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_RC4_128_SHA,TLS_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_RC4_128_SHA
I1005 16:35:23.732]       --tls-min-version string                                  Minimum TLS version supported. Possible values: VersionTLS10, VersionTLS11, VersionTLS12
I1005 16:35:23.733]       --tls-private-key-file string                             File containing the default x509 private key matching --tls-cert-file.
I1005 16:35:23.733]       --tls-sni-cert-key namedCertKey                           A pair of x509 certificate and private key file paths, optionally suffixed with a list of domain patterns which are fully qualified domain names, possibly with prefixed wildcard segments. If no domain patterns are provided, the names of the certificate are extracted. Non-wildcard matches trump over wildcard matches, explicit domain patterns trump over extracted names. For multiple key/certificate pairs, use the --tls-sni-cert-key multiple times. Examples: "example.crt,example.key" or "foo.crt,foo.key:*.foo.com,foo.com". (default [])
I1005 16:35:23.733]   -v, --v Level                                                 log level for V logs
I1005 16:35:23.733]       --vmodule moduleSpec                                      comma-separated list of pattern=N settings for file-filtered logging
I1005 16:35:23.733] 
I1005 16:35:23.733] panic: Get https://10.63.240.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.733] 
I1005 16:35:23.734] goroutine 1 [running]:
I1005 16:35:23.734] main.main()
I1005 16:35:23.734] 	/go/src/github.com/kubernetes-incubator/metrics-server/cmd/metrics-server/metrics-server.go:39 +0x13b
I1005 16:35:23.734] 
I1005 16:35:23.734] ENDLOG for container kube-system:metrics-server-v0.3.1-66499c78f6-xcqgr:metrics-server
I1005 16:35:23.903] Oct  5 16:35:23.903: INFO: Logs of kube-system/metrics-server-v0.3.1-66499c78f6-xcqgr:metrics-server-nanny on node gke-e2e-32483-7506d-default-pool-eb898fb6-dpdq
I1005 16:35:23.903] Oct  5 16:35:23.903: INFO:  : STARTLOG
I1005 16:35:23.903] ERROR: logging before flag.Parse: I1005 16:23:17.076832       1 pod_nanny.go:65] Invoked by [/pod_nanny --config-dir=/etc/config --cpu=40m --extra-cpu=0.5m --memory=35Mi --extra-memory=4Mi --threshold=5 --deployment=metrics-server-v0.3.1 --container=metrics-server --poll-period=300000 --estimator=exponential --minClusterSize=5]
I1005 16:35:23.904] ERROR: logging before flag.Parse: I1005 16:23:17.077000       1 pod_nanny.go:81] Watching namespace: kube-system, pod: metrics-server-v0.3.1-66499c78f6-xcqgr, container: metrics-server.
I1005 16:35:23.904] ERROR: logging before flag.Parse: I1005 16:23:17.077128       1 pod_nanny.go:82] storage: MISSING, extra_storage: 0Gi
I1005 16:35:23.904] ERROR: logging before flag.Parse: I1005 16:23:17.168223       1 pod_nanny.go:109] cpu: 40m, extra_cpu: 0.5m, memory: 35Mi, extra_memory: 4Mi
I1005 16:35:23.904] ERROR: logging before flag.Parse: I1005 16:23:17.168259       1 pod_nanny.go:138] Resources: [{Base:{i:{value:40 scale:-3} d:{Dec:<nil>} s:40m Format:DecimalSI} ExtraPerNode:{i:{value:5 scale:-4} d:{Dec:<nil>} s: Format:DecimalSI} Name:cpu} {Base:{i:{value:36700160 scale:0} d:{Dec:<nil>} s:35Mi Format:BinarySI} ExtraPerNode:{i:{value:4194304 scale:0} d:{Dec:<nil>} s:4Mi Format:BinarySI} Name:memory}]
I1005 16:35:23.904] ERROR: logging before flag.Parse: E1005 16:23:47.168740       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.905] ERROR: logging before flag.Parse: E1005 16:24:17.168500       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.905] ERROR: logging before flag.Parse: E1005 16:24:18.169367       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.905] ERROR: logging before flag.Parse: E1005 16:24:49.170062       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.905] ERROR: logging before flag.Parse: E1005 16:25:20.170768       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.905] ERROR: logging before flag.Parse: E1005 16:25:27.260586       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:25:51.171532       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:26:22.172316       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:26:37.261063       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:26:53.261117       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:27:24.261867       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.906] ERROR: logging before flag.Parse: E1005 16:27:47.261605       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:27:55.361074       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:28:26.361714       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:28:57.262141       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:28:57.362520       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:29:28.363211       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.907] ERROR: logging before flag.Parse: E1005 16:29:59.363923       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:30:07.262638       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:30:30.364641       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:31:01.365415       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:31:17.263189       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:31:32.366191       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.908] ERROR: logging before flag.Parse: E1005 16:32:03.366844       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.909] ERROR: logging before flag.Parse: E1005 16:32:27.263713       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.909] ERROR: logging before flag.Parse: E1005 16:32:34.367765       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.909] ERROR: logging before flag.Parse: E1005 16:33:05.368494       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.909] ERROR: logging before flag.Parse: E1005 16:33:36.369187       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.909] ERROR: logging before flag.Parse: E1005 16:33:37.264046       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.910] ERROR: logging before flag.Parse: E1005 16:34:07.370074       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.910] ERROR: logging before flag.Parse: E1005 16:34:38.370905       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.910] ERROR: logging before flag.Parse: E1005 16:34:47.264504       1 nanny_lib.go:87] timed out waiting for the condition
I1005 16:35:23.910] ERROR: logging before flag.Parse: E1005 16:35:09.371821       1 reflector.go:205] k8s.io/autoscaler/addon-resizer/nanny/kubernetes_client.go:107: Failed to list *v1.Node: Get https://10.63.240.1:443/api/v1/nodes?resourceVersion=0: dial tcp 10.63.240.1:443: i/o timeout
I1005 16:35:23.910] 
I1005 16:35:23.910] ENDLOG for container kube-system:metrics-server-v0.3.1-66499c78f6-xcqgr:metrics-server-nanny
```

**Comment by saad-ali** [comment]
SIG networking can you help[ investigate the kube-dns failure.
/sig networking

**Comment by saad-ali** [comment]
SIG instrumentation can you please investigate the metrics-server panic.
CC @kubernetes/sig-instrumentation-bugs

**Comment by MrHohn** [comment]
> I1005 16:35:23.301] panic: counter cannot decrease in value

Regarding the posted kube-dns error, that is a known issue: https://github.com/kubernetes/dns/issues/195. But it is a consequential failure triggered by dnsmasq container restarts, rather than the real root cause. Will take a deeper look.

**Comment by MrHohn** [comment]
Various logs indicate kube-dns pod failed to talk to outside --- k8s apiserver, stackdriver, etc.
```
# kubedns container
I1005 21:44:13.668] I1005 21:42:35.536661       1 dns.go:219] Waiting for [endpoints services] to be initialized from apiserver...
I1005 21:44:13.668] E1005 21:42:35.537428       1 reflector.go:201] k8s.io/dns/pkg/dns/dns.go:189: Failed to list *v1.Endpoints: Get https://10.11.240.1:443/api/v1/endpoints?resourceVersion=0: dial tcp 10.11.240.1:443: i/o timeout
I1005 21:44:13.668] E1005 21:42:35.537688       1 reflector.go:201] k8s.io/dns/pkg/dns/dns.go:192: Failed to list *v1.Service: Get https://10.11.240.1:443/api/v1/services?resourceVersion=0: dial tcp 10.11.240.1:443: i/o 

# prometheus-to-sd
I1005 21:44:14.233] I1005 21:33:55.079884       1 main.go:115] Running prometheus-to-sd, monitored target is kubedns localhost:10054
I1005 21:44:14.233] W1005 21:35:25.083307       1 stackdriver.go:82] Error while fetching metric descriptors for kubedns: Get https://test-monitoring.sandbox.googleapis.com/v3/projects/kube-gke-upg-1-1-1-2-upg-clu/metricDescriptors?alt=json&filter=metric.type+%3D+starts_with%28%22container.googleapis.com%2Finternal%2Faddons%2Fkubedns%22%29: dial tcp 108.177.111.81:443: i/o timeout
I1005 21:44:14.233] E1005 21:35:55.089785       1 stackdriver.go:58] Error while sending request to Stackdriver Post https://test-monitoring.sandbox.googleapis.com/v3/projects/kube-gke-upg-1-1-1-2-upg-clu/timeSeries?alt=json: dial tcp 74.125.69.81:443: i/o timeout

```

**Comment by MrHohn** [comment]
Humm weirdly I failed to reproduce this --- creating a GKE alpha cluster with the latest k8s build works for me.

**Comment by MrHohn** [comment]
Did some debugging on my dev cluster, seems like non-hostnetwork pods lost the ability to talk to outside. Checked on the newly deployed netd daemonset (enabled on alpha cluster) and found some errors that maybe relevant:
```
$ kubectl logs netd-dvfgl -n kube-system
...
I1005 23:22:03.152077       1 main.go:51] netd started
E1005 23:22:03.180950       1 config.go:208] failed to append rule [-m comment --comment ip-masq: ensure nat POSTROUTING directs all non-LOCAL destination traffic to our custom IP-MASQ chain -m addrtype ! --dst-type LOCAL -j IP-MASQ] in table nat chain POSTROUTING: running [/sbin/iptables -t nat -C POSTROUTING -m comment --comment ip-masq: ensure nat POSTROUTING directs all non-LOCAL destination traffic to our custom IP-MASQ chain -m addrtype ! --dst-type LOCAL -j IP-MASQ --wait]: exit status 2: iptables v1.6.1: Couldn't load target `IP-MASQ':No such file or directory

Try `iptables -h' or 'iptables --help' for more information.
E1005 23:22:03.181277       1 network_config_controller.go:72] found an error for Masquerade: running [/sbin/iptables -t nat -C POSTROUTING -m comment --comment ip-masq: ensure nat POSTROUTING directs all non-LOCAL destination traffic to our custom IP-MASQ chain -m addrtype ! --dst-type LOCAL -j IP-MASQ --wait]: exit status 2: iptables v1.6.1: Couldn't load target `IP-MASQ':No such file or directory

Try `iptables -h' or 'iptables --help' for more information.
 when ensuring {{nat POSTROUTING true 0xc42004e480} [[-m comment --comment ip-masq: ensure nat POSTROUTING directs all non-LOCAL destination traffic to our custom IP-MASQ chain -m addrtype ! --dst-type LOCAL -j
IP-MASQ]] 0xc42004e480}
```

Looping in @jingax10 @cadmuxe

**Comment by MrHohn** [comment]
Seems likely because the `IP-MASQ` chain is not appended completely, so outgoing packet with pod IP doesn't properly get SNAT to the node IP, hence gets dropped on the GCE fabric layer.

**Comment by jingax10** [comment]
It sounds like a rule ordering issue in https://github.com/GoogleCloudPlatform/netd/blob/master/pkg/config/config_masquerade.go#L33.

The first rule tries to jump to a custom chain, which is not created at the moment. I'm surprised that reconcile loop does not catch it in the next run.

Looking into the code and will send the fix tonight.

**Comment by jingax10** [comment]
@cadmuxe already fixed the issue in https://github.com/GoogleCloudPlatform/netd/pull/46

Waiting for GKE changes to be approved.

**Comment by jingax10** [comment]
We've fixed netd and rebuilt v0.18 and then made changes to GKE to pick up the new image.

The tests turned green since 10/09/2018 08:16.

Thereby, I closed the issue.

/close
